src/mutation.ts

```diff
@@ -116,6 +116,11 @@ export class Mutation {
 
       if (Number.isSafeInteger(num)) {
         return num;
+      } else {
+        const dblVal = buf.readDoubleBE();
+        if (!Number.isNaN(Number(dblVal)) && Number(dblVal) === dblVal) {
```

Comment:
What are you evaluating with `Number(dblVal) === dblVal`? Is that attempting to check the type of dblVal? If so `typeof(dblVal) === "number"` seems more natural.